crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -718,6 +756,174 @@ impl<F: Clone, Ctx: PaymentMethodRetrieve>
+                                logger::error!(error=?err);
+                                None
+                            },
+                            Some,
+                        )
+                    }
+                    _ => None,
+                });
+
+            // Hashed Cardbin to check whether or not this payment should be blocked.
+            let cardbin_hash = payment_data
+                .payment_method_data
+                .as_ref()
+                .and_then(|pm_data| match pm_data {
+                    api_models::payments::PaymentMethodData::Card(card) => {
+                        crypto::HmacSha512::sign_message(
+                            &crypto::HmacSha512,
+                            merchant_secret.as_bytes(),
+                            card.card_number.clone().get_card_isin().as_bytes(),
+                        )
+                        .change_context(errors::ApiErrorResponse::InternalServerError)
+                        .attach_printable("error in card bin hash creation")
+                        .map_or_else(
+                            |err| {
+                                logger::error!(error=?err);
+                                None
+                            },
+                            Some,
+                        )
+                    }
+                    _ => None,
+                });
+
+            // Hashed Extended Cardbin to check whether or not this payment should be blocked.
+            let extended_cardbin_hash =
+                payment_data
+                    .payment_method_data
+                    .as_ref()
+                    .and_then(|pm_data| match pm_data {
+                        api_models::payments::PaymentMethodData::Card(card) => {
+                            crypto::HmacSha512::sign_message(
+                                &crypto::HmacSha512,
+                                merchant_secret.as_bytes(),
+                                card.card_number.clone().get_extended_card_bin().as_bytes(),
+                            )
+                            .change_context(errors::ApiErrorResponse::InternalServerError)
+                            .attach_printable("error in extended card bin hash creation")
+                            .map_or_else(
+                                |err| {
+                                    logger::error!(error=?err);
+                                    None
+                                },
+                                Some,
+                            )
+                        }
+                        _ => None,
+                    });
+
+            let cardbin_encoded_hash = cardbin_hash.map(|id| consts::BASE64_ENGINE.encode(id));
+
+            let extended_cardbin_encoded_hash =
+                extended_cardbin_hash.map(|id| consts::BASE64_ENGINE.encode(id));
+
+            let fingerprint_encoded_hash =
+                fingerprint_hash.map(|id| consts::BASE64_ENGINE.encode(id));
```

Comment:
Why are we Base64 encoding the SHA-512 hash? Unless there's a specific reason, use `hex::encode` to get a string representation, cuz Base64 encoded data is needlessly longer.